### DISCUSSION TITLE
Tweak Volume Title Sort Order - Prioritise Parentheses

### DISCUSSION THREAD

**Discussion by falo2k** [gotson/komga#1701] [Ideas]
A byproduct of the title naming for volumes other than Volume 1 is that you can end up with sequential volumes being separated by other items in the listings.  For example, if I have the 1940, 2011, and 2016 volumes of Batman, they would be listed as:
- Batman
- Batman (2011)
- Batman (2016)

Because volumes such as Batman & Robin, Batman & The Outsiders, etc. exist, as do a number of two digit year series like Batman '66 and Batman '89, those volumes appear in between Batman and Batman (2011).  i.e. You end up with:

- Batman
- Batman & All the Other Characters We Can Sell A Book On Because it Has Batman
- Batman '66
- Batman (2011)
- Batman (2016)
- Batman - Titles That have a hyphen breaking the subtitle
- Batman / And Then the Slashes

I assume this is because ( is further down the sort order than other punctuation characters.  Would it be possible to either sort on the raw title (without any decorations like volume / year) or to amend the sort order to put parentheses up front when sorting punctuation characters?

Extra ignorable bit:
It may also be worth thinking about subtitles following semi-colons (e.g. Batman: Bane) as those get sorted after any other books (: sorts after the space character) which can look a bit odd, though realistically you can't chase down all the cases as there's no standard for punctuation in titles (e.g. you get plenty of "Batman / Character" but also "Batman/Character") so I suppose it ends up being subjective as to what order they should be for those.

**Comment by gotson** [comment]
You should use the _Sort Title_ for series for such use case. ComicInfo.xml does not provide such information, so the title is used as sort title.

**Reply by falo2k** [reply]
I don't see a separate option for "Title", only "Name" in the sort orders.  The issue here is that "Title" is amended by Komga to integrate the year when series.json is available (according to https://komga.org/docs/guides/scan-analysis-refresh#import-metadata-generated-by-mylar).

![image](https://github.com/user-attachments/assets/eac93c53-f614-48d5-9d87-9902fe659d26)

**Reply by gotson** [reply]
try to edit a series, you will see the 2 fields Title and Title Sort

<img width="579" alt="image" src="https://github.com/user-attachments/assets/d90478f3-0338-4d70-889b-f18b4f4a6efa">

**Reply by falo2k** [reply]
Then ideally there should be a setting that doesn't just put the same altered title in that field when parsing the metadata.  At this stage, I'm not going to go and manually tweak system generated metadata manually across my library for instances of multiple volumes.

**Reply by gotson** [reply]
> Then ideally there should be a setting that doesn't just put the same altered title in that field when parsing the metadata.

this is a `ComicInfo.xml` limitation

**Comment by falo2k** [comment]
I don't follow how that's the case - at no point does ComicInfo.xml or series.json concatenate "Batman" and "(2016)" - that has to be done on import, so is a choice made by the workflow.  Is the solution here to be subtler with the generation of "Sort Title" and not use a character that gets sorted after other common characters.

![image](https://github.com/user-attachments/assets/27527664-e4c2-45a6-a3f9-c4b81e5e5028)

**Reply by gotson** [reply]
https://komga.org/docs/guides/scan-analysis-refresh/#series-metadata

> The Series and Volume elements will be used to overwrite the title of the Series, in the form <Series> (<Volume>), or just <Series> if the Volume element is not present, if the Volume is 1, or if Append volume to series title is disabled. If multiple values are present, the most frequent value from all books will be used.

**Reply by falo2k** [reply]
Yes, I'm aware it's making those changes.  That's a choice of the import workflow.  The issue is in making that choice to use parentheses for the decoration for both the title and the "sort title" you introduce this sorting issue when looking at a library.  It should be possible to either change the character used in sort title, or make it consistent across ALL books (so as not to leave Volume 1 out in the cold, unsorted and alone).
